### ISSUE TITLE
getDriver() not defined

### ISSUE THREAD

**Issue by andrekat** [mirko-pagliai/cakephp-database-backup#9]
Hello! 

I have recently installed and configured your backup plugin. Somehow it wont work .. I am using the BackupExport to make my own export (in customShell). 

Running the shell causes following error:

Exception: Call to undefined method Cake\Database\Connection::getDriver() in [/var/www/xxx/public_html/one/vendor/mirko-pagliai/cakephp-database-backup/src/BackupTrait.php, line 119]
2017-07-24 13:18:12 Error: [Error] Call to undefined method Cake\Database\Connection::getDriver()

Any ideas?

**Comment by mirko-pagliai** [comment]
Hello @andrekat Last version (2.1.1)? Php version?
Can you post the full stack?

**Comment by andrekat** [comment]
I think I have finally found the problem ... you are using the method Connection::getDriver() - this method has been added on CakePHP 3.4 - I am using 3.3, which does not offer getDriver method, instead it is using Connection::driver() method. 

I will try to update my application to the latest CakePHP version and give you afterwards feedback!

**Comment by mirko-pagliai** [comment]
No, the `Connection::getDriver()` is from my plugin, not from Cake.

**Comment by andrekat** [comment]
Oh ok ... 

Stack:

`Exception: Call to undefined method Cake\Database\Connection::getDriver() in [/var/www/xxx/public_html/one/vendor/mirko-pagliai/cakephp-database-backup/src/BackupTrait.php, line 119]
2017-07-24 13:18:12 Error: [Error] Call to undefined method Cake\Database\Connection::getDriver()
Stack Trace:
#0 /var/www/xxx/public_html/one/vendor/mirko-pagliai/cakephp-database-backup/src/Utility/BackupExport.php(96): DatabaseBackup\Utility\BackupExport->getDriver(Object(Cake\Database\Connection))
#1 /var/www/xxx/public_html/one/src/Shell/BackupShell.php(68): DatabaseBackup\Utility\BackupExport->__construct()
#2 /var/www/xxx/public_html/one/src/Shell/BackupShell.php(47): App\Shell\BackupShell->backupDatabase()
#3 [internal function]: App\Shell\BackupShell->main()
#4 /var/www/xxx/public_html/one/vendor/cakephp/cakephp/src/Console/Shell.php(472): call_user_func_array(Array, Array)
#5 /var/www/xxx/public_html/one/vendor/cakephp/cakephp/src/Console/ShellDispatcher.php(227): Cake\Console\Shell->runCommand(Array, true, Array)
#6 /var/www/xxx/public_html/one/vendor/cakephp/cakephp/src/Console/ShellDispatcher.php(182): Cake\Console\ShellDispatcher->_dispatch(Array)
#7 /var/www/xxx/public_html/one/vendor/cakephp/cakephp/src/Console/ShellDispatcher.php(128): Cake\Console\ShellDispatcher->dispatch(Array)
#8 /var/www/xxx/public_html/one/bin/cake.php(34): Cake\Console\ShellDispatcher::run(Array)
#9 {main}
`

**Comment by mirko-pagliai** [comment]
Apologizes. It's like you say.
Before upgrading, make a test. Edit the file `vendor/mirko-pagliai/cakephp-database-backup/src/BackupTrait.php`, line 119:

        $className = $this->getClassShortName($connection->driver());

Just to see if this is the only problem of compability of the version.

**Comment by andrekat** [comment]
No problem .. It is working now 👍  on both sides (3.3 (edit) and 3.4 (without editing) .

Thanks for your fast response!

Regards,

**Comment by mirko-pagliai** [comment]
Unfortunately that is not the only method used by the plugin and only available with the latest version of cakephp.

**Comment by mirko-pagliai** [comment]
Release 2.1.2
https://github.com/mirko-pagliai/cakephp-database-backup/releases/tag/2.1.2

> fixed composer.json: the plugin requires at least version 3.4 of CakePHP.

**Comment by mirko-pagliai** [comment]
Today I have released 2 new versions:
- `2.1.4`, which requires at least CakePHP 3.4;
- `2.2.0`, which requires at least CakePHP 3.5.

For anyone who has the same problem: I created the [cakephp3.2](https://github.com/mirko-pagliai/cakephp-database-backup/tree/cakephp3.2) branch, which contains code compatible with all versions above 3.2. In essence, it retains the methods that have been deprecated in the latest versions.

To install this version:

    $ composer require --prefer-dist mirko-pagliai/cakephp-database-backup:dev-cakephp3.2
